docs/docs-beta/docs/guides/tbd/running-local-ui-development.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Running Dagster locally
+description: How to run Dagster on your local machine.
+---
+
+# Running Dagster locally
+
+In this guide, we'll walk you through how to run Dagster on your local machine using the `dagster dev` command.  The `dagster dev` command launches the Dagster UI and the Dagster daemon, allowing you to start a full deployment of Dagster from the command line.
+
+## Locating your code
+
+Before starting local development, you need to tell Dagster how to find the Python code containing your assets and jobs.
+
+<Tabs>
+  <TabItem value="module" label="From a module">
+    Dagster can load Python modules as code locations.
+    <CodeExample filePath="guides/tbd/definitions.py" language="python" title="my_module/__init__.py" />
+    We can use the `-m` argument to supply the name of the module to start a Dagster instance loaded with our definitions:
+    ```shell
+    dagster dev -m my_module
+    ```
+
+  </TabItem>
+  <TabItem value="without-args" label="Without command line arguments">
+    To load definitions from a module without supplying the `-m` command line argument, you can use a `pyproject.toml` file. This file, included in all Dagster example projects, contains a `tool.dagster` section where you can supply the `module_name`:
+    <CodeExample filePath="guides/tbd/pyproject.toml" language="toml" title="pyproject.toml" />
+
+
+  </TabItem>
+  <TabItem value="file" label="From a file">
+    Dagster can load a file directly as a code location.
+    <CodeExample filePath="guides/tbd/definitions.py" language="python" title="definitions.py" />
+    Given the above file, we can use the `-f` argument to supply the name of the file to start a Dagster instance loaded with our definitions:
+    ```shell
+    dagster dev -f defs.py
+    ```
+
+    *Note:* We don't recommend using the `-f` argument for production deployments, to avoid a whole class of Python import errors.
```

Comment:
::: note
We don't recommend using the `-f` argument for production deployments.

:::